src/aleph/vm/hypervisors/qemu_confidential/qemuvm.py

```diff
@@ -0,0 +1,122 @@
+        # hardware_resources.published ports -> not implemented at the moment
+        # hardware_resources.seconds -> only for microvm
+
+        # TODO : ensure this is ok at launch
+        sev_info = secure_encryption_info()
+        if sev_info is None:
+            raise ValueError("Not running on an AMD SEV platform?")
+        godh = self.sev_dh_cert_file
+        launch_blob = self.sev_session_file
+
+        if not (godh.is_file() and launch_blob.is_file()):
+            raise FileNotFoundError("Missing guest owner certificates, cannot start the VM.`")
+        args = [
+            self.qemu_bin_path,
+            "-enable-kvm",
+            "-nodefaults",
+            "-m",
+            str(self.mem_size_mb),
+            "-smp",
+            str(self.vcpu_count),
+            "-drive",
+            f"if=pflash,format=raw,unit=0,file={self.ovmf_path},readonly=on",
+            "-drive",
+            f"file={self.image_path},media=disk,if=virtio,format=qcow2",
+            # To debug you can pass gtk or curses instead
+            "-display",
+            "none",
+            "--no-reboot",  # Rebooting from inside the VM shuts down the machine
+            # Listen for commands on this socket
+            "-monitor",
+            f"unix:{self.monitor_socket_path},server,nowait",
+            # Listen for commands on this socket (QMP protocol in json). Supervisor use it to send shutdown or start
+            # command
+            "-qmp",
+            f"unix:{self.qmp_socket_path},server,nowait",
+            # Tell to put the output to std fd, so we can include them in the log
+            "-nographic",
+            "-serial",
+            "stdio",
+            "--no-reboot",  # Rebooting from inside the VM shuts down the machine
+            "-S",
+            # Confidential options
+            "-object",
+            f"sev-guest,id=sev0,policy={self.sev_policy},cbitpos={sev_info.c_bit_position},"
+            f"reduced-phys-bits={sev_info.phys_addr_reduction},"
+            f"dh-cert-file={godh},session-file={launch_blob}",
+            "-machine",
+            "confidential-guest-support=sev0",
+            # Linux kernel 6.9 added a control on the RDRAND function to ensure that the random numbers generation
+            # works well, on Qemu emulation for confidential computing the CPU model us faked and this makes control
+            # raise an error and prevent boot. Passing the argument --cpu host instruct the VM to use the same CPU
+            # model than the host thus the VM's kernel knows which method is used to get random numbers (Intel and
+            # AMD have different methods) and properly boot.
+            "-cpu",
+            "host",
+            # Uncomment following for debug
+            # "-serial", "telnet:localhost:4321,server,nowait",
+            # "-snapshot",  # Do not save anything to disk
```

Comment:
Some of that commands are so similar with the standard `QemuVM` class, so maybe will be better to extract to a method and avoid code duplication.